drivers/input/input_cap1203.c

```diff
@@ -65,24 +69,21 @@ static int cap1203_process(const struct device *dev)
 	struct cap1203_data *data = dev->data;
 	int r;
 	uint8_t input;
-	uint16_t col;
-	bool pressed;
+	uint8_t single_input_state;
 
 	r = i2c_reg_read_byte_dt(&config->i2c, REG_INPUT_STATUS, &input);
 	if (r < 0) {
 		return r;
 	}
 
-	pressed = !!input;
-	if (input & BIT(0)) {
-		col = 0;
-	}
-	if (input & BIT(1)) {
-		col = 1;
-	}
-	if (input & BIT(2)) {
-		col = 2;
+	for (uint8_t i = 0; i < TOUCH_INPUT_COUNT; i++) {
+		single_input_state = input & BIT(i);
+		if (single_input_state != (data->prev_input_state & BIT(i))) {
+			input_report_key(dev, config->input_codes[i], (single_input_state >> i),
```

Comment:
No need to shift back the value, `input_report_key` already applies a `!!` to it: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/input/input.h#L82, fine to leave `single_input_state` in here.